crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -122,70 +122,75 @@ impl GeneratedAcir {
     pub(crate) fn call_black_box(
         &mut self,
         func_name: BlackBoxFunc,
-        mut inputs: Vec<FunctionInput>,
+        inputs: &[Vec<FunctionInput>],
         constants: Vec<FieldElement>,
+        output_count: usize,
     ) -> Result<Vec<Witness>, InternalError> {
-        intrinsics_check_inputs(func_name, &inputs)?;
+        let input_count = inputs.iter().fold(0usize, |sum, val| sum + val.len());
+        intrinsics_check_inputs(func_name, input_count);
+        intrinsics_check_outputs(func_name, output_count);
 
-        let output_count = black_box_expected_output_size(func_name)?;
         let outputs = vecmap(0..output_count, |_| self.next_witness_index());
 
         // clone is needed since outputs is moved when used in blackbox function.
         let outputs_clone = outputs.clone();
 
         let black_box_func_call = match func_name {
             BlackBoxFunc::AND => {
-                BlackBoxFuncCall::AND { lhs: inputs[0], rhs: inputs[1], output: outputs[0] }
+                BlackBoxFuncCall::AND { lhs: inputs[0][0], rhs: inputs[1][0], output: outputs[0] }
             }
             BlackBoxFunc::XOR => {
-                BlackBoxFuncCall::XOR { lhs: inputs[0], rhs: inputs[1], output: outputs[0] }
+                BlackBoxFuncCall::XOR { lhs: inputs[0][0], rhs: inputs[1][0], output: outputs[0] }
             }
-            BlackBoxFunc::RANGE => BlackBoxFuncCall::RANGE { input: inputs[0] },
-            BlackBoxFunc::SHA256 => BlackBoxFuncCall::SHA256 { inputs, outputs },
-            BlackBoxFunc::Blake2s => BlackBoxFuncCall::Blake2s { inputs, outputs },
-            BlackBoxFunc::HashToField128Security => {
-                BlackBoxFuncCall::HashToField128Security { inputs, output: outputs[0] }
+            BlackBoxFunc::RANGE => BlackBoxFuncCall::RANGE { input: inputs[0][0] },
+            BlackBoxFunc::SHA256 => BlackBoxFuncCall::SHA256 { inputs: inputs[0].clone(), outputs },
+            BlackBoxFunc::Blake2s => {
+                BlackBoxFuncCall::Blake2s { inputs: inputs[0].clone(), outputs }
             }
+            BlackBoxFunc::HashToField128Security => BlackBoxFuncCall::HashToField128Security {
+                inputs: inputs[0].clone(),
+                output: outputs[0],
+            },
             BlackBoxFunc::SchnorrVerify => BlackBoxFuncCall::SchnorrVerify {
-                public_key_x: inputs[0],
-                public_key_y: inputs[1],
+                public_key_x: inputs[0][0],
+                public_key_y: inputs[1][0],
                 // Schnorr signature is an r & s, 32 bytes each
-                signature: inputs[2..66].to_vec(),
-                message: inputs[66..].to_vec(),
```

Comment:
Mind explaining how these relate?

Reply:
Do you mean in the schnorr algorithm in general? Due to the previous way we were handling inputs to black box funcs we only enabled a single variable input (in this case it is the message that is being verified with this signature). We had to index a flattened inputs array to fetch the respective signature and message, while now they are separate into their own function input vectors.